### ISSUE TITLE
[bug]: react hooks gets executed two times

### ISSUE THREAD

**Issue by kambleaa007** [reactjs/reactjs.org#3587]
as i clicked on button only once,
![image](https://user-images.githubusercontent.com/19818679/111965030-48ff4980-8b1b-11eb-9a7e-dec919d105eb.png)
call stack
![image](https://user-images.githubusercontent.com/19818679/111965088-59afbf80-8b1b-11eb-913d-b18ec839d252.png)
 after continue, it gets called again without clicking on button

gets called again
![image](https://user-images.githubusercontent.com/19818679/111965182-78ae5180-8b1b-11eb-8f2a-387c7f79f270.png)
so updating value not once but twice
here 
![image](https://user-images.githubusercontent.com/19818679/111965249-8a8ff480-8b1b-11eb-87f4-995de2f04c20.png)
you can see value 7 instead of 5

reproduction steps:
goto url http://localhost:3000/myApp
code url https://github.com/kambleaa007/TrainingReactJS

the same code works for first time, not from second time onwards
trying for first time this works,
![image](https://user-images.githubusercontent.com/19818679/111965586-e5c1e700-8b1b-11eb-99a6-6e98b1cf4442.png)
then
in debug
![image](https://user-images.githubusercontent.com/19818679/111965634-f5413000-8b1b-11eb-879e-3295768d9641.png)
here worked
![image](https://user-images.githubusercontent.com/19818679/111965685-04c07900-8b1c-11eb-8b68-504a12084831.png)

for first time
not problem in my code logic as it works first time second time onwards error persist

**Comment by eps1lon** [comment]
Duplicate of https://github.com/facebook/react/issues/21048
